scripts/benchmarks_local.py

```diff
@@ -0,0 +1,404 @@
+                "generatelayoutonly",
+                "/p:LibrariesConfiguration=Release"
+            ] + additional_args
+    RunCommand(generate_layout_command, verbose=True).run(os.path.join(repo_path, "src/tests"))
+
+def get_run_artifact_path(parsed_args: Namespace, run_type: RunType, commit: str) -> str:
+    return os.path.join(parsed_args.artifact_storage_path, f"{run_type.name}-{commit}-{parsed_args.os}-{parsed_args.architecture}-{parsed_args.framework}")
+
+# Try to generate all of a single runs dependencies at once to save time
+def generate_all_runtype_dependencies(parsed_args: Namespace, repo_path: str, commit: str, force_regenerate: bool = False):
+    getLogger().info(f"Generating dependencies for {' '.join(map(str, parsed_args.run_type_names))} run types in {repo_path} and storing in {parsed_args.artifact_storage_path}.")
+    
+    if check_for_runtype_specified(parsed_args, [RunType.CoreRun]):
+        dest_dir = os.path.join(get_run_artifact_path(parsed_args, RunType.CoreRun, commit), "Core_Root")
+        
+        if force_regenerate or not os.path.exists(dest_dir):
+            build_runtime_dependency(parsed_args, repo_path)
+            generate_layout(parsed_args, repo_path)
+            # Store the corerun in the artifact storage path
+            core_root_path = os.path.join(repo_path, "artifacts", "tests", "coreclr", f"{parsed_args.os}.{parsed_args.architecture}.Release", "Tests", "Core_Root")
+            shutil.rmtree(dest_dir, ignore_errors=True)
+            copy_directory_contents(core_root_path, dest_dir)
+            shutil.rmtree(os.path.join(repo_path, "artifacts"), ignore_errors=True)
+        else:
+            getLogger().info(f"CoreRun already exists in {dest_dir}. Skipping generation.")
+
+    if check_for_runtype_specified(parsed_args, [RunType.MonoInterpreter, RunType.MonoJIT]):
+        dest_dir_mono_interpreter = os.path.join(get_run_artifact_path(parsed_args, RunType.MonoInterpreter, commit), "dotnet_mono")
+        dest_dir_mono_jit = os.path.join(get_run_artifact_path(parsed_args, RunType.MonoJIT, commit), "dotnet_mono")
+
+        if force_regenerate or not os.path.exists(dest_dir_mono_interpreter) or not os.path.exists(dest_dir_mono_jit):
+            build_runtime_dependency(parsed_args, repo_path, "clr+mono+libs")
+            build_runtime_dependency(parsed_args, repo_path, "libs.pretest", additional_args=['-testscope', 'innerloop', '/p:RuntimeFlavor=mono', f"/p:RuntimeArtifactsPath={os.path.join(repo_path, 'artifacts', 'bin', 'mono', f'{parsed_args.os}.{parsed_args.architecture}.Release')}"])
+
+            # Create the mono-dotnet
+            src_dir = os.path.join(repo_path, "artifacts", "bin", "runtime", f"net8.0-{parsed_args.os}-Release-{parsed_args.architecture}")
+            dest_dir = os.path.join(repo_path, "artifacts", "bin", "testhost", f"net8.0-{parsed_args.os}-Release-{parsed_args.architecture}", "shared", "Microsoft.NETCore.App", "8.0.0")
```

Comment:
I think for all the `net8.0`/`8.0.0`/... (here and and all the below) we should create constant(s) for easy future changes.